### PR TITLE
(WIP) batch http->https fixes

### PR DESCRIPTION
**Pull request by bfontaine**
I checked all the stable URLs (using `brew fetch` and ensuring there were no errors/warnings), plus some of the homepages.

- [x] stable
- [x] devel
- [x] HEAD
- [x] Homepages

### CODE REVIEW

**Comment by bfontaine** [comment]
I killed the build.

File: Library/Formula/pcre2.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class Pcre2 < Formula
   desc "Perl compatible regular expressions library with a new API"
-  homepage "http://www.pcre.org/"
+  homepage "https://pcre.org/"
```
Comment (vszakats) [inline Library/Formula/pcre2.rb]: The `www.`-less URL is fine, but sadly the https:// authentication fails with a domain mismatch.

File: Library/Formula/zeromq.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class Zeromq < Formula
   desc "High-performance, asynchronous messaging library"
-  homepage "http://www.zeromq.org/"
+  homepage "https://www.zeromq.org/"
```
Comment (vszakats) [inline Library/Formula/zeromq.rb]: Same here, it's a domain mismatch with the certificate.

**Comment by bfontaine** [comment]
Thanks @vszakats, I removed these (and killed the build).

File: Library/Formula/ansible.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 class Ansible < Formula
   desc "Automate deployment, configuration, and upgrading"
-  homepage "http://www.ansible.com/home"
-  url "http://releases.ansible.com/ansible/ansible-1.9.2.tar.gz"
+  homepage "https://www.ansible.com/home"
```
Comment (vszakats) [inline Library/Formula/ansible.rb]: Noticed another CA mismatch in `homepage` above.

File: Library/Formula/flashrom.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class Flashrom < Formula
   desc "Identify, read, write, verify, and erase flash chips"
-  homepage "http://flashrom.org/"
+  homepage "https://flashrom.org/"
```
Comment (vszakats) [inline Library/Formula/flashrom.rb]: This one uses a CACert certificate and for another domain.

File: Library/Formula/gearman.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 class Gearman < Formula
   desc "Application framework to farm out work to other machines or processes"
-  homepage "http://gearman.org/"
+  homepage "https://gearman.org/"
```
Comment (vszakats) [inline Library/Formula/gearman.rb]: Couldn't connect to this one via `https://` just now.

File: Library/Formula/liblqr.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@
 class Liblqr < Formula
   desc "C/C++ seam carving library"
   homepage 'http://liblqr.wikidot.com/'
```
Comment (vszakats) [inline Library/Formula/liblqr.rb]: This one can be updated to `https://`, which will redirect to plaintext (for non-logged-in users I guess).

File: Library/Formula/libvbucket.rb [hunk 7]
```diff
@@ -3,7 +3,7 @@
 class Libvbucket < Formula
   desc "Utility library providing mapping to virtual buckets"
   homepage 'http://couchbase.com/develop/c/current'
```
Comment (vszakats) [inline Library/Formula/libvbucket.rb]: This one can be updated to `https://`.

File: Library/Formula/little-cms2.rb [hunk 8]
```diff
@@ -1,6 +1,6 @@
 class LittleCms2 < Formula
   desc "Color management engine supporting ICC profiles"
-  homepage "http://www.littlecms.com/"
+  homepage "https://www.littlecms.com/"
```
Comment (vszakats) [inline Library/Formula/little-cms2.rb]: There is a certificate domain mismatch here.

File: Library/Formula/openconnect.rb [hunk 9]
```diff
@@ -1,6 +1,6 @@
 class Openconnect < Formula
   desc "Open client for Cisco AnyConnect VPN"
-  homepage "http://www.infradead.org/openconnect.html"
+  homepage "https://www.infradead.org/openconnect.html"
```
Comment (vszakats) [inline Library/Formula/openconnect.rb]: Domain mismatch and CACert certificate here.

File: Library/Formula/parrot.rb [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
 class Parrot < Formula
   desc "Open source virtual machine (for Perl6, et al.)"
-  homepage "http://www.parrot.org/"
+  homepage "https://www.parrot.org/"
```
Comment (vszakats) [inline Library/Formula/parrot.rb]: Its wildcard certificate expired in 2011.

File: Library/Formula/u-boot-tools.rb [hunk 11]
```diff
@@ -1,6 +1,6 @@
 class UBootTools < Formula
   desc "Universal boot loader"
-  homepage "http://www.denx.de/wiki/U-Boot/"
+  homepage "https://www.denx.de/wiki/U-Boot/"
```
Comment (vszakats) [inline Library/Formula/u-boot-tools.rb]: This one uses a self-signed certificate.

**Comment by bfontaine** [comment]
@vszakats Thank you for all your comments regarding the mismatched certificates, in fact I checked on my side with `curl` and when I pushed I saw you commented on all the formulae I just removed :wink: 

This PR is a first step, I’m not trying to fix *all* formulae (that can be) with it. I wrote a Ruby script to parse [HTTPS-Everywhere’s rules](https://github.com/EFForg/https-everywhere/tree/master/src/chrome/content/rules) and applied it on Homebrew but it doesn’t support all kind of rules for now, and I got a lot of false positives as you saw.

File: Library/Formula/libvbucket.rb (hunk 7 above)
Comment (vszakats) [inline Library/Formula/libvbucket.rb]: It is mixed-content, but these are usually accepted. (should be fixed upstream eventually)

**Comment by vszakats** [comment]
Using HTTPS Everywhere rules is great, thumbs up! I now understand why these were included. I was about to recommend it a while ago, only their rules seemed complex to parse.

**Comment by bfontaine** [comment]
Yeah I remember someone mentionned these in a PR, maybe that was you. The rules didn’t seem hard to parse so I tried, but in practice there are a lot of weird cases and it didn’t get as smooth as I was expected.

You can find [the parser here](https://github.com/bfontaine/tohttps.rb/blob/brew/tohttps.rb) and [the script here](https://github.com/bfontaine/tohttps.rb/blob/brew/brew-fix-http.rb) (warning: these are quick & dirty).

**Comment by vszakats** [comment]
Ah, that's great, nice job.

**Comment by bfontaine** [comment]
Any other comment before I merge this?

**Comment by mikemcquaid** [comment]
Do it :+1:
